src/core/public/application/types.ts

```diff
@@ -86,6 +86,12 @@ export interface App extends AppBase {
    * Takes precedence over chrome service visibility settings.
    */
   chromeless?: boolean;
+
+  /**
+   * Override the application's base routing path from `/app/${id}`.
+   * Must be unique across registered applications.
+   */
+  appBasePath?: string;
```

Comment:
I think we should use a different name since this can be conflated with `AppMountParams.appBasePath` which are not the same string. Maybe `appRoute`? Also can we make it clear in this comment that this string should not include the basePath from `http.basePath`.